content/exchange/artifacts/Windows.Hunter.Yara.LOLDrivers.yaml

```diff
@@ -0,0 +1,113 @@
+
+      LET driver_dirs = SELECT
+          lowcase(string=path_join(components=path_split(path=PathName)[:-1])) AS Folder
+      FROM wmi(
+          query = "SELECT PathName FROM Win32_SystemDriver",
+          namespace = "ROOT\\CIMV2"
+      )
+      GROUP BY Folder
+
+      LET files = SELECT * FROM foreach(
+        row=driver_dirs,
+        query={
+          SELECT OSPath, Name, Size
+          FROM glob(globs=Folder + "/*", nosymlink="True")
+          WHERE NOT IsDir AND NOT IsLink
+        })
+
+      LET vuln_hits = SELECT * FROM foreach(row=files, query={
+        SELECT OSPath, Rule, Tags, Meta,
+               String.Name AS YaraString,
+               String.Offset AS HitOffset,
+               "vulnerability" AS MatchType
+        FROM yara(rules=decoded_vuln_rule, files=OSPath)
```

Comment:
you can put the above in the export section then just call them with different Yara rules as the logic is exactly the same:

```sql
-- In export section
LET Scan(YaraRules, Type) = SELECT * FROM foreach(row=files, query={
        SELECT OSPath, Rule, Tags, Meta,
               String.Name AS YaraString,
               String.Offset AS HitOffset,
               Type AS MatchType
        FROM yara(rules=decoded_vuln_rule, files=OSPath)

-- In artifact source
SELECT * FROM  Scan(YaraRules=..., Type="vulnerability') 

```